component/src/main/java/io/siddhi/extension/map/csv/sourcemapper/CSVSourceMapper.java

```diff
@@ -148,7 +148,7 @@ public class CSVSourceMapper extends SourceMapper {
     private boolean isCustomMappingEnabled = false;
     private StreamDefinition streamDefinition;
     private boolean failOnUnknownAttribute;
-    private Character delimiter;
+    private Object delimiter;
```

Comment:
@AnuGayan Can we store the delimiter as a `String`? or could there be a performance impact if we store a character as a string and do `csvFormat = CSVFormat.Builder.create().setDelimiter(delimiter.toString()).build();` ?

If we store delimiter as a string, we'll be able to avoid doing: 
- `delimiterValue.length() > 1` under `init` method.
- `delimiter instanceof Character` under `getFormatter` method.

Reply:
When we work with tab delimiter, it won't work if we pass it as a String, so we have to use char for that

Reply:
Won't it be possible to store as `private String delimiter;`

Then do the following in `init`:
```
if (TAB_DElIMITER_INPUT.equals(delimiterValue)) {
            this.delimiter = TAB_DElIMITER; // Have TAB_DElIMITER as String
} else {
            this.delimiter = delimiterValue
}
```

And finally do the following in `getFormatter`:
```
private CSVFormat getFormatter(String delimiter) {
        return TAB_DElIMITER.equals(delimiter) ? CSVFormat.TDF :
                   CSVFormat.Builder.create().setDelimiter(delimiter).build();
}
```

Is there anything that I'm missing here?